src/main/java/com/epam/test_generator/services/StepService.java

```diff
@@ -123,4 +124,33 @@ public void removeStep(Long suitId, Long caseId, Long stepId) {
 
         caseVersionDAO.save(caze);
     }
+
+    /**
+     * This service method is specialized for 'adding' 'editing' and 'deleting' a list of steps
+     *
+     * For selecting required action uses "action" field from DTO object
+     *
+     * @param steps array list of steps
+     */
+    public void cascadeUpdateSteps(long suitId, long caseId, List<StepDTO> steps) {
+        for (StepDTO stepDTO : steps) {
+            Action action = stepDTO.getAction();
+            if (action != null) {
+                switch (action) {
+                    case UPDATE:
+                        updateStep(suitId, caseId, stepDTO.getId(), stepDTO);
+                        break;
+                    case CREATE:
+                        addStepToCase(suitId, caseId, stepDTO);
+                        break;
+                    case DELETE:
+                        removeStep(suitId, caseId, stepDTO.getId());
+                        break;
+                    default:
+                        // Do nothing
```

Comment:
I'd recommend throwing an exception or adding logs to a 'default' statement.

Reply:
This is a true customer requirement and behavior 
[quote]In a payload each step in collection will have an action ("update","delete","create"). 
Ignore steps with empty actions[/quote]
Unfortunately, there is no logging yet in the project.